pkg/machine/ignition/ignition.go

```diff
@@ -879,3 +913,17 @@ func DefaultReadyUnitFile() parser.UnitFile {
 	u.Add("Install", "RequiredBy", "default.target")
 	return *u
 }
+
+func GetRosettaActivationUnitFile() *parser.UnitFile {
+    rosettaUnit := parser.NewUnitFile()
+    rosettaUnit.Add("Unit", "Description", "Activates Rosetta if necessary")
+    rosettaUnit.Add("Unit", "After", "sshd.socket sshd.service")
+	rosettaUnit.Add("Service", "Type", "oneshot")
+	rosettaUnit.Add("Service", "RemainAfterExit", "yes")
+    rosettaUnit.Add("Service", "ExecStartPre","mount -t virtiofs -o context=system_u:object_r:container_runtime_exec_t:s0 rosetta /mnt")
+	rosettaUnit.Add("Service", "ExecStart","/usr/local/bin/rosetta-activation.sh")
+    rosettaUnit.Add("Service", "ExecStartPost",`find /proc/sys/fs/binfmt_misc -type f -name 'qemu-*' -exec sh -c 'echo -1 > {}' \;`)
```

Comment:
This seems to unregister all qemu- emulators but rosetta only support x86-64 so it should only remove this one.
Also I don't know the kernel handles registering the same arch twice? I would logically speaking think one would first remove qemu and than add rosetta.

Reply:
Agree. And I removed only x86_64 and replaced the order of the commands.